Source/Core/Core/HLE/HLE_OS.cpp

```diff
@@ -75,14 +88,79 @@ void HLE_write_console()
   NOTICE_LOG(OSREPORT, "%08x->%08x| %s", LR, PC, SHIFTJISToUTF8(report_message).c_str());
 }
 
-std::string GetStringVA(u32 strReg)
+// Log (v)dprintf message if fd is 1 (stdout) or 2 (stderr)
+void HLE_LogDPrint(bool has_va_list)
 {
-  std::string ArgumentBuffer;
-  u32 ParameterCounter = strReg + 1;
-  u32 FloatingParameterCounter = 1;
+  NPC = LR;
+
+  if (GPR(3) != 1 && GPR(3) != 2)
+    return;
+
+  std::string report_message = GetStringVA(4, has_va_list);
+  NOTICE_LOG(OSREPORT, "%08x->%08x| %s", LR, PC, SHIFTJISToUTF8(report_message).c_str());
+}
+
+// Log dprintf message
+//  -> int dprintf(int fd, const char* format, ...);
+void HLE_LogDPrint()
+{
+  HLE_LogDPrint(false);
+}
+
+// Log vdprintf message
+//  -> int vdprintf(int fd, const char* format, va_list ap);
+void HLE_LogVDPrint()
+{
+  HLE_LogDPrint(true);
+}
+
+// Log (v)fprintf message if FILE is stdout or stderr
+void HLE_LogFPrint(bool has_va_list)
+{
+  NPC = LR;
+
+  // The structure FILE is implementation defined.
+  // Both libogc and Dolphin SDK seem to store the fd at the same address.
+  int fd = -1;
+  if (PowerPC::HostIsRAMAddress(GPR(3)) && PowerPC::HostIsRAMAddress(GPR(3) + 0xF))
```

Comment:
Not sure I can here. How should I consider libogc case? AFAIK, it would be the same for both GC and Wii. Considering the different members of FILE structures it would be very unlucky to find 1 or 2 while the rest is mainly filled of zero or pointer addresses most of the time.

To be honest, I'd rather have more logs with false positives than not enough because we're too strict.